clang/lib/Sema/TreeTransform.h

```diff
@@ -7239,6 +7239,13 @@ QualType TreeTransform<Derived>::TransformAttributedType(TypeLocBuilder &TLB,
       });
 }
 
+template <typename Derived>
+QualType TreeTransform<Derived>::TransformCountAttributedType(
+    TypeLocBuilder &TLB, CountAttributedTypeLoc TL) {
+  // TODO
+  llvm_unreachable("Unexpected TreeTransform for CountAttributedType");
```

Comment:
Hmmm this might be problematic sooner rather than later; the attribute is only allowed in C mode so we don't have to worry about needing this for template instantiation, but we do use `TreeTransform` in C for OpenMP: https://github.com/llvm/llvm-project/blob/d2a9df2c8ffd21fd52fbd8199a191d10078f41af/clang/lib/Sema/SemaOpenMP.cpp#L5591

Reply:
@AaronBallman thanks for letting me know. I just implemented `TransformCountAttributedType`. I'll see if I can fine a test case to exercise this.